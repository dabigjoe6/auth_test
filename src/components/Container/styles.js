import {colors} from '../../config';

const styles = {
  container: {
    flex: 1,
    paddingHorizontal: 30,
    paddingTop: 90,
  },
  title: {
    fontSize: 22,
    fontWeight: '900',
    marginBottom: 30,
    color: colors.primary,
  },
};

export default styles;
