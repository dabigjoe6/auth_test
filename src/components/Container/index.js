import React from 'react';
import {View, Text} from 'react-native';

import styles from './styles';

const Container = ({children, title}) => {
  return (
    <View testID="Container" style={styles.container}>
      <Text testId="ContainerTitle" style={styles.title}>
        {title}
      </Text>
      {children}
    </View>
  );
};

export default Container;
