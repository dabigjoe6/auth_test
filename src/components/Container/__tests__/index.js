import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';

import Container from '../index';

describe('Container component', () => {
  const child = <Text>Custom Child</Text>;

  it('should render without issues', () => {
    const component = renderer.create(<Container>{child}</Container>);
    const instance = component.root;
    expect(
      instance.findAll(
        (comp) => comp.props.testID === 'Container' && comp.type === 'View',
      ).length,
    ).toBe(1);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render title', () => {
    const component = renderer.create(
      <Container title="Test title">{child}</Container>,
    );
    const instance = component.root;
    const viewInstance = instance.findAll(
      (comp) => comp.props.testID === 'Container' && comp.type === 'View',
    )[0];

    const text = viewInstance.findAll(
      (comp) => comp.props.testId === 'ContainerTitle' && comp.type === 'Text',
    )[0];

    expect(text.children).toEqual([instance.props.title]);
  });
});
