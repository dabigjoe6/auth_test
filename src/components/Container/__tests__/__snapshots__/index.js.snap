// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render without issues 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 30,
      "paddingTop": 90,
    }
  }
  testID="Container"
>
  <Text
    style={
      Object {
        "color": "#74B3CE",
        "fontSize": 22,
        "fontWeight": "900",
        "marginBottom": 30,
      }
    }
    testId="ContainerTitle"
  />
  <Text>
    Custom Child
  </Text>
</View>
`;
